app/scripts/controllers/main.js

```diff
@@ -174,9 +180,7 @@ angular.module('moodCatApp')
         }
 
         //Send the mesage to the server
-        chatService.sendChatMessage($scope.chatMessage, $scope.room.id).then(function(res) {
-          $scope.messages.push(res);
-        });
+        chatService.sendChatMessage($scope.chatMessage, $scope.room.id);
```

Comment:
First of all , why doesnt this work? The response sent from a new backend should include a proper message id.
If it's really impossible to leave this as it was, then you should immediately fetch the newest messages for the best experience.

```js
  chatService.sendChatMessage($scope.chatMessage, $scope.room.id).then(this.fetchMessages.bind(this));
```